processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -555,19 +592,32 @@ private File makeIndexFiles(
       dimConversions.add(Maps.<String, IntBuffer>newHashMap());
     }
 
+    Map<String, Integer> nullStringIndexMap = Maps.<String, Integer>newHashMap();
```

Comment:
I think this is always going to be 0 when a dimension has null values. If so then it could just be a Set, and the MMappedIndexRowIterable could use a constant array like `int[] NULL_VALUE = {0}`